cli/f3.go

```diff
@@ -39,25 +46,161 @@ var (
+						tablewriter.Col(from),
+						tablewriter.Col(to),
 					)
 					for _, participant := range miners {
+						addr, err := address.NewIDAddress(participant.MinerID)
+						if err != nil {
+							return fmt.Errorf("converting miner ID to address: %w", err)
+						}
+
 						tw.Write(map[string]interface{}{
-							idColumn:   participant.MinerID,
-							fromColumn: participant.FromInstance,
-							ToColumn:   participant.FromInstance + participant.ValidityTerm,
+							miner: addr,
+							from:  participant.FromInstance,
+							to:    participant.FromInstance + participant.ValidityTerm,
 						})
 					}
 					return tw.Flush(cctx.App.Writer)
 				},
 			},
+			{
+				Name:    "certs",
+				Aliases: []string{"c"},
+				Usage:   "Manages interactions with F3 finality certificates.",
+				Subcommands: []*cli.Command{
+					{
+						Name: "get",
+						Usage: "Gets an F3 finality certificate to a given instance ID, " +
+							"or the latest certificate if no instance is specified.",
+						ArgsUsage: "[instance]",
+						Flags: []cli.Flag{
+							f3FlagOutput,
+						},
+						Before: func(cctx *cli.Context) error {
+							if count := cctx.NArg(); count > 1 {
+								return fmt.Errorf("too many arguments: expected at most 1 but got %d", count)
+							}
+							return nil
+						},
+						Action: func(cctx *cli.Context) error {
+							api, closer, err := GetFullNodeAPIV1(cctx)
+							if err != nil {
+								return err
+							}
+							defer closer()
+
+							// Get the certificate, either for the given instance or the latest if no
+							// instance is specified.
+							var cert *certs.FinalityCertificate
+							if cctx.Args().Present() {
+								var instance uint64
+								instance, err = strconv.ParseUint(cctx.Args().First(), 10, 64)
+								if err != nil {
+									return fmt.Errorf("parsing instance: %w", err)
+								}
+								cert, err = api.F3GetCertificate(cctx.Context, instance)
+							} else {
+								cert, err = api.F3GetLatestCertificate(cctx.Context)
+							}
+							if err != nil {
+								return fmt.Errorf("getting finality certificate: %w", err)
+							}
+							if cert == nil {
+								_, _ = fmt.Fprintln(cctx.App.Writer, "No certificate.")
+								return nil
+							}
+
+							return outputFinalityCertificate(cctx, cert)
+						},
+					},
+					{
+						Name: "list",
+						Usage: `Lists a range of F3 finality certificates.
+
+A range may optionally be specified as the first argument to indicate 
+inclusive range of 'from' and 'to' instances in following notation:
+'<from>..<to>'. Either <from> or <to> may be omitted, but not both.
+An omitted instance is interpreted as the latest instance.
+
+If no range is specified, certificates are listed in descending order
+of instance, starting from the latest until the limit is reached, i.e.
+the default range of '..0'.
+
+Examples:
+  * '5..': from instance 5 to the latest instance.
+  * '..0': from the latest instance to instance 0.
```

Comment:
IMO, this is kind of confusing (especially coming from rust, git, etc.). I'd expect `..5` to mean "0..5`, not `reverse(5..latest)`. If we want to provide a way to reverse, I'd just provide a flag.

Reply:
It _is_ the more compact way of specifying and (imo) intuitive: 2..3 is in reverse order of 3..2. A missing to or from, e.g. ..2 can only be <from latest> to 2, because there is already a number associated to instance zero:`0`.

If you really really hate it I will change it.

Reply:
I can live with it but I think it's going to confuse everyone/everyone who sees it because it breaks convention with every language I know that treats `..x` to mean `min_value..x`. Git, rust, and even go/python (via `:x` and `x:`).

Also, I want to consider how this interacts with limit. E.g., give me the top 10 certs in reverse order is written as `lotus f3 c list ..0` which is... kind of non-obvious. There is no way to say "give me the top 10 certs in-order", although I don't know if that's necessary. Now that I look at this more closely:

1. I'd expect/want `lotus f3 c list` to return the latest 10? certs in reverse order (newest first).
2. I'd actually expect the default limit for any `lotus f3 c list ARG` command to be _infinity_ unless otherwise specified to match, e.g., git.

But maybe I'm just making this all more complicated than it needs to be.

Reply:
> I'd expect/want lotus f3 c list to return the latest 10? certs in reverse order (newest first).

The default does this right? i.e. `lotus f3 c list` wold return top 10 certs from latest to latest-10th instance. I think you meant "oldest first"?

> I'd actually expect the default limit for any lotus f3 c list ARG command to be infinity unless otherwise specified to match, e.g., git.

My thought was that since the certs are (hopefully) generated every 2 minutes or so an unlimited default might make up a very verbose output. Happy to change the default.

Thanks @Stebalien, I will change this to introduce a dedicated order flag and use the more popular rage notation.

Reply:
Just now I have defaulted to no limit. You probably have a better sense of what makes a good default. 

Happy to limit it and add "N certificates elided" or leave it as unlimited. 

I guess we can always change the default if folks start complaining?

Reply:
I meant "no limit _if_ the user specifies a range". Also, don't assume I know what users expect: I live in Emacs.